src/backend/common/MemoryManagerImpl.hpp

```diff
@@ -130,16 +124,18 @@ void MemoryManager<T>::setMaxMemorySize() {
     }
 }
 
-template<typename T>
-MemoryEventPair MemoryManager<T>::alloc(const size_t bytes, bool user_lock) {
-    MemoryEventPair ptr = {nullptr, detail::Event()};
+af_buffer_info MemoryManager::alloc(const size_t bytes, bool user_lock) {
```

Comment:
We should probably move the common::MemoryManager class to the src/api/c directory now that it must use the C API to function correctly.